### PR TITLE
Changed MaxFrameSize feature description

### PR DESCRIPTION
**Pull request by kovakimy**

### CODE REVIEW

File: doc/mediasdk-man.md [hunk 1]
```diff
@@ -4139,16 +4139,16 @@ The application can attach this extended buffer to the [mfxVideoParam](#mfxVideo
...
 `MotionVectorsOverPicBoundaries` | When set to OFF, no sample outside the picture boundaries and no sample at a fractional sample position for which the sample value is derived using one or more samples outside the picture boundaries is used for inter prediction of any sample.<br><br>When set to ON, one or more samples outside picture boundaries may be used in inter prediction.<br><br>See the [CodingOptionValue](#CodingOptionValue) enumerator for values of this option.
-`MaxFrameSizeI` | Same as [mfxExtCodingOption2](#mfxExtCodingOption2)`::MaxFrameSize` but affects only I-frames.
-`MaxFrameSizeP` | Same as [mfxExtCodingOption2](#mfxExtCodingOption2)`::MaxFrameSize` but affects only P-frames.
+`MaxFrameSizeI` | Same as [mfxExtCodingOption2](#mfxExtCodingOption2)`::MaxFrameSize` but affects only I-frames. Maximum allowed frame size per profile/level will be calculated in driver and be used when MaxFrameSizeI and MaxFrameSizeP are both set to 0. Feature, when MaxFrameSizeI is not specified and MaxFrameSizeP is specified, is unsupported.
```
Comment (dmitryermilov) [inline doc/mediasdk-man.md]: IMO, "in driver" is bad wording. MediaSDK manual is hw agnostic.

File: doc/mediasdk-man.md [hunk 2]
```diff
@@ -3978,7 +3978,7 @@ The application can attach this extended buffer to the [mfxVideoParam](#mfxVideo
...
 `IntRefQPDelta` | Specifies QP difference for inserted intra MBs. This is signed value in [-51, 51] range. This parameter is valid during initialization and runtime.
-`MaxFrameSize` | Specify maximum encoded frame size in byte. This parameter is used in VBR based bitrate control modes and ignored in others. The SDK encoder tries to keep frame size below specified limit but minor overshoots are possible to preserve visual quality. This parameter is valid during initialization and runtime. It is recommended to set MaxFrameSize to 5x-10x target frame size (`(TargetKbps*1000)/(8* FrameRateExtN/FrameRateExtD)`) for I frames and 2x-4x target frame size for P frames.
+`MaxFrameSize` | Specify maximum encoded frame size in byte. This parameter is used in VBR based bitrate control modes and ignored in others(CBR, CQP). The SDK encoder tries to keep frame size below specified limit but minor overshoots are possible to preserve visual quality. This parameter is valid during initialization and runtime. It is recommended to set MaxFrameSize to 5x-10x target frame size (`(TargetKbps*1000)/(8* FrameRateExtN/FrameRateExtD)`) for I frames and 2x-4x target frame size for P/B frames.
```
Comment (dmitryermilov) [inline doc/mediasdk-man.md]: Is "(CBR, CQP)" mentioning really needed? What about look ahead BRC,  ICQ, AVBR, etc ?

File: doc/mediasdk-man.md (hunk 1 above)
Comment (stasefremov) [reply doc/mediasdk-man.md]: Please re-formulate this "Feature, when MaxFrameSizeI is not specified and MaxFrameSizeP is specified, is unsupported.". Does it mean that MaxFrameSizeI must be set if MaxFrameSizeI is set?

Comment (kovakimy) [reply doc/mediasdk-man.md]: @stasefremov It means that MaxFrameSizeI must be set if MaxFrameSizeP is set

Comment (stasefremov) [reply doc/mediasdk-man.md]: Then I think description "MaxFrameSizeI must be set if MaxFrameSizeP is set" is much more clear than "Feature, when MaxFrameSizeI is not specified and MaxFrameSizeP is specified, is unsupported." :)

Comment (stasefremov) [reply doc/mediasdk-man.md]: "Maximum allowed frame size per profile/level will be calculated in driver and be used when MaxFrameSizeI and MaxFrameSizeP are both set to 0.". I think AVC spec limitations will still be applied even when respective MaxFrameSizeX is set. E.g. if MaxFrameSizeI/MaxFrameSizeP is set to value bigger than spec limitation. Also spec limitation will be applied to P-frames if MaxFrameSizeI is set and MaxFrameSizeP isn't set. So current statement looks too narrow.

Comment (kovakimy) [reply doc/mediasdk-man.md]: @stasefremov Do you mean, we need to mention, if both values ​​are not set or exceed the max value, then the MaxFrameSizeX will be default?

Comment (stasefremov) [reply doc/mediasdk-man.md]: "if both values ​​are not set or exceed the max value, then the MaxFrameSizeX will be default?" It's much closer but still not there. There is still word "both" which looks incorrect to me. I don't think we apply AVC spec limitation for frame size ONLY if BOTH MaxFrameSizeI/MaxFrameSizeP are not set.  If MaxFrameSizeI is set and MaxFrameSizeP is not set, we still apply AVC spec limitation for P-frames, right?

File: doc/mediasdk-man.md [hunk 3]
```diff
@@ -4139,16 +4139,16 @@ The application can attach this extended buffer to the [mfxVideoParam](#mfxVideo
...
-`MaxFrameSizeI` | Same as [mfxExtCodingOption2](#mfxExtCodingOption2)`::MaxFrameSize` but affects only I-frames. Maximum allowed frame size per profile/level will be calculated in driver and be used when MaxFrameSizeI and MaxFrameSizeP are both set to 0. MaxFrameSizeI must be set if MaxFrameSizeP is set.
-`MaxFrameSizeP` | Same as [mfxExtCodingOption2](#mfxExtCodingOption2)`::MaxFrameSize` but affects only P/B-frames. If MaxFrameSizeP equals 0, MaxFrameSizeI will be used for all frame types. Maximum allowed frame size per profile/level will be calculated in driver and be used when MaxFrameSizeI and MaxFrameSizeP are both set to 0.
+`MaxFrameSizeI` | Same as [mfxExtCodingOption2](#mfxExtCodingOption2)`::MaxFrameSize` but affects only I-frames. MaxFrameSizeI must be set if MaxFrameSizeP is set. If MaxFrameSizeI and MaxFrameSizeP are not specified or greater than spec limitation, spec limitation will be applied to the values.
```
Comment (stasefremov) [inline doc/mediasdk-man.md]: I would formulate as "If MaxFrameSizeI is not specified or greater than spec limitation, spec limitation will be applied to the sizes of I-frames".

File: doc/mediasdk-man.md [hunk 4]
```diff
@@ -4139,16 +4139,16 @@ The application can attach this extended buffer to the [mfxVideoParam](#mfxVideo
...
+`MaxFrameSizeP` | Same as [mfxExtCodingOption2](#mfxExtCodingOption2)`::MaxFrameSize` but affects only P/B-frames. If MaxFrameSizeP equals 0, MaxFrameSizeI will be used for all frame types. If MaxFrameSizeI and MaxFrameSizeP are not specified or greater than spec limitation, spec limitation will be applied to the values.
```
Comment (stasefremov) [inline doc/mediasdk-man.md]: I would formulate like this "If MaxFrameSizeP equals 0, the SDK sets MaxFrameSizeP equal to MaxFrameSizeI. If MaxFrameSizeP is not set or greater than spec limitation, spec limitation will be applied to the sizes of P/B frames".

File: doc/mediasdk-man.md [hunk 5]
```diff
@@ -3978,7 +3978,7 @@ The application can attach this extended buffer to the [mfxVideoParam](#mfxVideo
...
 `IntRefQPDelta` | Specifies QP difference for inserted intra MBs. This is signed value in [-51, 51] range. This parameter is valid during initialization and runtime.
-`MaxFrameSize` | Specify maximum encoded frame size in byte. This parameter is used in VBR based bitrate control modes and ignored in others(CBR, CQP). The SDK encoder tries to keep frame size below specified limit but minor overshoots are possible to preserve visual quality. This parameter is valid during initialization and runtime. It is recommended to set MaxFrameSize to 5x-10x target frame size (`(TargetKbps*1000)/(8* FrameRateExtN/FrameRateExtD)`) for I frames and 2x-4x target frame size for P/B frames.
+`MaxFrameSize` | Specify maximum encoded frame size in byte. This parameter is used in VBR based bitrate control modes and ignored in others. The SDK encoder tries to keep frame size below specified limit but minor overshoots are possible to preserve visual quality. This parameter is valid during initialization and runtime. It is recommended to set MaxFrameSize to 5x-10x target frame size (`(TargetKbps*1000)/(8* FrameRateExtN/FrameRateExtD)`) for I frames and 2x-4x target frame size for P/B frames.
```
Comment (stasefremov) [inline doc/mediasdk-man.md]: Do we have "VBR based bitrate control modes" defined somewhere in the spec?

**Comment by dmitryermilov** [comment]
LGTM.
@stasefremov , @DenWolf , @oakasatkina , please review

**Pull request merged**